data_model/derive/src/id.rs

```diff
@@ -1,120 +1,121 @@
-#![allow(clippy::expect_used, clippy::mixed_read_write_in_expression)]
+#![allow(clippy::str_to_string, clippy::mixed_read_write_in_expression)]
 
-use proc_macro::TokenStream;
+use proc_macro2::TokenStream;
+use proc_macro_error::abort;
 use quote::quote;
-use syn::{
-    parse::{Parse, ParseStream},
-    Attribute, Field, Generics, Ident, Token, TypePath, Visibility,
-};
-
-pub struct IdInput {
-    ident: Ident,
-    id_type: TypePath,
-}
+use syn::parse_quote;
 
-impl Parse for IdInput {
-    fn parse(input: ParseStream) -> syn::Result<Self> {
-        let attrs = input.call(Attribute::parse_outer)?;
-        let id_type = parse_id_attribute(&attrs);
+fn derive_identifiable(input: &syn::ItemStruct) -> TokenStream {
+    let name = &input.ident;
+    let (impl_generics, ty_generics, where_clause) = input.generics.split_for_impl();
+    let (id_type, id_expr) = get_id_type(input);
 
-        let _vis = input.parse::<Visibility>()?;
-        let _struct_token = input.parse::<Token![struct]>()?;
-        let ident = input.parse()?;
-        let _generics = input.parse::<Generics>()?;
-        let content;
-        let _brace_token = syn::braced!(content in input);
-        let _struct_fields = content.parse_terminated::<Field, Token![,]>(Field::parse_named)?;
+    quote! {
+        impl #impl_generics Identifiable for #name #ty_generics #where_clause {
```

Comment:
if I use `iroha_data_model::Identifiable` I don't know how to use this path in the `iroha_data_model` because there it should be `crate::Identifiable`